### PR TITLE
move gradle parallel option to env section

### PR DESCRIPTION
**Pull request by imanushin**
Simplify build via moving gradle parallel option to the end of workflow.

Please don't merge until I compare the performance of builds.

### CODE REVIEW

File: .github/workflows/pre-merge.yaml [hunk 1]
```diff
@@ -113,4 +113,8 @@ jobs:
         with:
           java-version: 14
       - name: Build and compile test snippets
-        run: ./gradlew test -x ":detekt-gradle-plugin:test" -Pcompile-test-snippets=true --parallel
+        run: ./gradlew test -x ":detekt-gradle-plugin:test" -Pcompile-test-snippets=true
+
+env:
+   GRADLE_OPTS: --parallel
```
Comment (cortinico) [inline .github/workflows/pre-merge.yaml:119]: This should probably be merged with the other `GRADLE_OPTS` we already have. Moreover it should be: `-Dorg.gradle.parallel=true`

Comment (arturbosch) [reply .github/workflows/pre-merge.yaml:119]: `--parallel` is a shorthand for `-Dorg.gradle.parallel=true`. Are there any reasons to prefer the long form?

Comment (imanushin) [reply .github/workflows/pre-merge.yaml:119]: Oups, I forgot about the [same thing before](https://github.com/detekt/detekt/pull/2808#issuecomment-647002151), sorry.

I prefer the longer name (e.g. ```-Dorg.gradle.parallel=true```) just for explicit purpose. However this is just my non-objective opinion, nothing more.

**Comment by imanushin** [comment]
After compairing agains another pull request - looks like the build time isn't worse:

https://github.com/detekt/detekt/pull/2934
```
@github-actions
Code Coverage / publish-code-coverage (pull_request) Successful in 11m
Details
@github-actions
Pre Merge Checks / gradle (ubuntu-latest, 8) (pull_request) Successful in 1m
Details
@github-actions
Validate Gradle Wrapper / Validation (pull_request) Successful in 6s
Details
@github-actions
Pre Merge Checks / gradle (ubuntu-latest, 11) (pull_request) Successful in 1m
Details
@github-actions
Pre Merge Checks / gradle (ubuntu-latest, 14) (pull_request) Successful in 1m
Details
@github-actions
Pre Merge Checks / gradle (macos-latest, 8) (pull_request) Successful in 22m
Details
@github-actions
Pre Merge Checks / gradle (macos-latest, 11) (pull_request) Successful in 3m
Details
@github-actions
Pre Merge Checks / gradle (macos-latest, 14) (pull_request) Successful in 2m
Details
@github-actions
Pre Merge Checks / gradle (windows-latest, 8) (pull_request) Successful in 3m
Details
@github-actions
Pre Merge Checks / gradle (windows-latest, 11) (pull_request) Successful in 3m
Details
@github-actions
Pre Merge Checks / gradle (windows-latest, 14) (pull_request) Successful in 7m
Details
@github-actions
Pre Merge Checks / verify-documentation (pull_request) Successful in 1m
Details
@github-actions
Pre Merge Checks / compile-test-snippets (pull_request) Successful in 45s
Details
@codecov
codecov/patch — Coverage not affected when comparing 4d0d8c9...fdd663f
Details
@codecov
codecov/project — 80.28% (+0.04%) compared to 4d0d8c9
Details
```

File: .github/workflows/pre-merge.yaml (hunk 1 above)
Comment (cortinico) [reply .github/workflows/pre-merge.yaml]: > Are there any reasons to prefer the long form?

Not really tbh. But it's probably better to have all the flags applied with the same style on the same line.

File: .github/workflows/codecoverage.yaml [hunk 2]
```diff
@@ -31,3 +31,6 @@ jobs:
         if: success()
         run: bash <(curl -s https://codecov.io/bash)
         shell: bash
+
+env:
+   GRADLE_OPTS: -Dorg.gradle.parallel=true
```
Comment (cortinico) [inline .github/workflows/codecoverage.yaml:36]: > This should probably be merged with the other GRADLE_OPTS we already have.

I don't know if you missed this part from my previous comment

Comment (imanushin) [reply .github/workflows/codecoverage.yaml:36]: Yes, this is my fault. Will fix, thank you.

**Review by schalkms — COMMENTED** [review]
Can we please check again, whether this change results in CI performance benefits? Thanks 😊

**Comment by BraisGabin** [comment]
How should we proceed here?

**Comment by schalkms** [comment]
> How should we proceed here?

If there are performance benefits, this PR should be merged. If not, then this shall be closed in my opinion.

**Comment by chao2zhang** [comment]
We already have `org.gradle.parallel=true` in `gradle.properties`, so this PR can be closed in my opinion.

Note: https://github.com/detekt/detekt/pull/3604 is already merged which overlaps with this PR.

**Pull request closed without merging**